assessment_factory/assessment_factory.py

```diff
@@ -1,65 +1,238 @@
+            'is_graded': self.is_graded,
+        }
+        frag = Fragment()
+        frag.add_content(loader.render_template('/public/html/assessment_factory.html', context))
+        frag.add_css(self.resource_string("public/css/vendors/jquery-ui.css"))
+        frag.add_css(self.resource_string("public/css/vendors/font-awesome.min.css"))
+        frag.add_css(self.resource_string("public/css/assessment_factory.css"))
+        frag.add_javascript(self.resource_string("public/js/src/assessment_factory.js"))
+        frag.initialize_js('AssessmentFactoryBlock', context)
         return frag
 
-    # TO-DO: change this handler to perform your own actions.  You may need more
-    # than one handler, or you may not need any handlers at all.
     @XBlock.json_handler
-    def increment_count(self, data, suffix=''):
-        """
-        An example handler, which increments the data.
-        """
-        # Just to show data coming in...
-        assert data['hello'] == 'world'
+    def submit_item(self, data, suffix=''):
+        item_id = data.get("item_id")
+        item_info = {
+            "item_id": item_id,
+            "category_id": data.get("category_id"),
+            "zone_id": data.get("zone_id"),
+            "style": data.get("style"),
+        }
+        self.item_state[item_id] = item_info;
+        return {
+            'result': 'success',
+        }
+
+    @XBlock.json_handler
+    def reset(self, data, suffix=''):
+        self.item_state = {}
+        self.incorrect_items = []
+        self.incorrect_items = []
+        self.retry_count = 3
+        self.is_graded = False
+        context = {
+            'studio_assignment': self.studio_assignment,
+            'item_state': self.item_state,
+            'retry_count': self.retry_count
+        }
+        return {
+            'ctx': context,
+        }
+
+    @XBlock.json_handler
+    def check_problem(self, data, suffix=''):
+        if self.item_state:
+            if self.retry_count > 0: 
+                data = self._get_correct_incorrect_items()
+                if data['incorrect_items'] != self.incorrect_items and data['correct_items'] != self.correct_items:
+                    self.retry_count -= 1
+                    self.incorrect_items = data['incorrect_items']
+                    self.correct_items = data['correct_items']
+                                
+        return {
+            'incorrect_items': data['incorrect_items'],
+            'correct_items': data['correct_items'],
+            'retry_count': self.retry_count
+        }
+
+    @XBlock.json_handler
+    def submit_problem(self, data, suffix=''):
+        if not self.is_graded:
+            grade = self._get_grade()
+            try:
+                self.runtime.publish(self, 'grade', {
+                    'value': grade,
+                    'max_value': self.weight,
+                })
+                self.is_graded = True
+            except NotImplementedError:
```

Comment:
aha, mogu provat vidit zašto su oni stavili, al generalno ako nemaš nekih očitih iznimki, onda ti i ne triba. ne vidim šta može bit neimplementirano tu